haystack/document_stores/in_memory/document_store.py

```diff
@@ -208,17 +208,22 @@ def bm25_retrieval(
         # get the last top_k indexes and reverse them
         top_docs_positions = np.argsort(docs_scores)[-top_k:][::-1]
 
+        # Negative values are possible under these conditions and should not be filtered out
+        negatives_are_valid = "Okapi" in self.bm25_algorithm.__name__ and not scale_score
```

Comment:
I would compare the type here or the exact name. Also I would expand the explanation a bit.